src/gui/optionsdialog.cpp

```diff
@@ -1866,29 +1872,29 @@ void OptionsDialog::webUIHttpsKeyChanged(const Path &path)
         (isKeyValid ? u"security-high"_s : u"security-low"_s), 24));
 }
 
-bool OptionsDialog::isWebUiEnabled() const
+bool OptionsDialog::isWebUIEnabled() const
 {
-    return m_ui->checkWebUi->isChecked();
+    return m_ui->checkWebUI->isChecked();
 }
 
-QString OptionsDialog::webUiUsername() const
+QString OptionsDialog::webUIUsername() const
 {
-    return m_ui->textWebUiUsername->text();
+    return m_ui->textWebUIUsername->text();
 }
 
-QString OptionsDialog::webUiPassword() const
+QString OptionsDialog::webUIPassword() const
 {
-    return m_ui->textWebUiPassword->text();
+    return m_ui->textWebUIPassword->text();
 }
 
 bool OptionsDialog::webUIAuthenticationOk()
 {
-    if (webUiUsername().length() < 3)
+    if (webUIUsername().length() < 3)
     {
         QMessageBox::warning(this, tr("Length Error"), tr("The Web UI username must be at least 3 characters long."));
         return false;
     }
-    if (!webUiPassword().isEmpty() && (webUiPassword().length() < 6))
+    if (!webUIPassword().isEmpty() && (webUIPassword().length() < 6))
```

Comment:
Shouldn't this be:
```c++
if (webUIPassword().length() < 6)
```
??

Reply:
> Shouldn't this be:
> 
> ```c++
> if (webUIPassword().length() < 6)
> ```
> 
> ??

No. We has no password in input field when editing settings (since it is unknown). Writing empty password to settings is prevented in other place of code.

Reply:
As far as I can tell `OptionsDialog::webUIAuthenticationOk()` is a way to force certain parameters on the admin/pass before closing the options window.
Currently, we don't enforce a non-empty password in the options window if the webui is enabled. Aka we don't tell the user to fix it. Only if the admin/password is short.
Maybe it is a bug that the calling sites of `webUIAuthenticationOk()` don't check if WebUI is enabled?

Reply:
> As far as I can tell ...

As I said above we don't know the password when editing the settings. Empty password just means "password was not changed" so nothing is written in settings:
https://github.com/qbittorrent/qBittorrent/blob/b7a0dee53d8bcacaab55cc35524c1a7f105a6053/src/gui/optionsdialog.cpp#L1323-L1324

Reply:
Let me rephrase my issue: If user has opened the preferences dialog, enabled the webui and not set a password*, we shouldn't let him close the dialog while the WebUI settings is checked. Silently failing is not a good approach in this case.

* I mean no password either on disk or in the input box.

Could this be implemented without much refactoring?

Reply:
Maybe a 3rd check?
```c++
if (isWebUIEnabled() && getWebUIPassword().isEmpty() && webUIPassword().isEmpty())
{
      QMessageBox::warning(this, tr("Unset Password"), tr("The WebUI password must be set."));
      return false;
}
```

Reply:
> Let me rephrase my issue: If user has opened the preferences dialog, enabled the webui and not set a password*, we shouldn't let him close the dialog while the WebUI setting is checked. Silently failing is not a good approach in this case.
> 
> *I mean no password either on disk or in the input box.

Ah, I get it now... I'll think about it.

Reply:
> Ah, I get it now... I'll think about it.

Well, I fixed it in GUI.
I don't know how best to do the same on the WebUI. I should at least send there information about whether there is password configured at the moment (aka `has_webui_password_configured`).

Reply:
> I don't know how best to do the same on the WebUI. I should at least send there information about whether there is password configured at the moment (aka `has_webui_password_configured`).

Well, seems there is no such problem for WebUI. After all, in order to change these settings from the WebUI, the user must be logged in, i.e. some password is configured in any case (or a temporary one is used).